src/server.c

```diff
@@ -3462,6 +3462,10 @@ int mustObeyClient(client *c) {
     return c->id == CLIENT_ID_AOF || c->flag.primary;
 }
 
+int canRedirectClient(client *c) {
+    return !server.cluster_enabled && server.primary_host && c->capa & CLIENT_CAPA_REDIRECT;
```

Comment:
The name of this function doesn't hint that it's only for standalone mode. (It always returns false in cluster mode.)

Can we make sure this function is only used in standalone mode and do an assert that cluster is disabled? I think it removes some confusion about cluster redirects.
 
```suggestion
    serverAssert(!server.cluster_enabled);
    return server.primary_host && c->capa & CLIENT_CAPA_REDIRECT;
```

Where we call it, we could check cluster mode before calling it, like

```
    if (!server.cluster_enabled && canRedirectClient(c)) { ... }
```

Reply:
Since `REDIRECT` is an error code only in standalone mode, I included the cluster_enabled check in `canRedirectClient`. I'm OK with your suggestion too.

Reply:
Ah, so Redirect in the name canRedirectClient refers only to the REDIRECT reply? We often refer to MOVED and ASK as redirects.

Reply:
I agree with @zuiderkwast that we can consider having a more explicit name e.g. `clientSupportStandAloneRedirect`or something